lib/providers/memory.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const LRU = require('lru-cache');
+const micromatch = require('micromatch');
+const yup = require('yup');
+
+const schema = yup.object().shape({
+  ttl: yup.number()
+});
+
+class Memory {
+  constructor(config) {
+    config = schema.validateSync(config, { stripUnknown: true });
+    this.name = 'memory';
```

Comment:
If we replace change tests to perform check using `instanceOf` instead of the name field check, can we remove name field completely from the providers?

Reply:
I thought the name field could be handy if the consumer wants to check which provider is being used. I think that the name field introduces very low noise for potential benefit, I found it very useful in the library-like scenario. What do you think?

Reply:
@ikovac that information is accessible through the reflection too (i.e. via `instanceOf` operator). It seems to me that the `name` field only introduces that same information in another place, so I can't see additional advantage.
It could be useful to export provider classes anyway, in case someone wants to inherit and extend their behavior.